x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/MountSearchableSnapshotRequest.java

```diff
@@ -0,0 +1,435 @@
+     *
+     * @return restore settings
+     */
+    public Settings settings() {
+        return this.settings;
+    }
+
+    /**
+     * Sets the list of index settings and index settings groups that shouldn't be restored from snapshot
+     */
+    public MountSearchableSnapshotRequest ignoreIndexSettings(String... ignoreIndexSettings) {
+        this.ignoreIndexSettings = ignoreIndexSettings;
+        return this;
+    }
+
+    /**
+     * Sets the list of index settings and index settings groups that shouldn't be restored from snapshot
+     */
+    public MountSearchableSnapshotRequest ignoreIndexSettings(List<String> ignoreIndexSettings) {
+        this.ignoreIndexSettings = ignoreIndexSettings.toArray(new String[ignoreIndexSettings.size()]);
+        return this;
+    }
+
+    /**
+     * Returns the list of index settings and index settings groups that shouldn't be restored from snapshot
+     */
+    public String[] ignoreIndexSettings() {
+        return ignoreIndexSettings;
+    }
+
+    /**
+     * Sets settings that should be added/changed for restored index
+     */
+    public MountSearchableSnapshotRequest indexSettings(Settings settings) {
+        this.indexSettings = settings;
+        return this;
+    }
+
+    /**
+     * Sets settings that should be added/changed for restored index
+     */
+    public MountSearchableSnapshotRequest indexSettings(Settings.Builder settings) {
+        this.indexSettings = settings.build();
+        return this;
+    }
+
+    /**
+     * Sets settings that should be added/changed for restored index
+     */
+    public MountSearchableSnapshotRequest indexSettings(String source, XContentType xContentType) {
+        this.indexSettings = Settings.builder().loadFromSource(source, xContentType).build();
+        return this;
+    }
+
+    /**
+     * Sets settings that should be added/changed for restored index
+     */
+    public MountSearchableSnapshotRequest indexSettings(Map<String, Object> source) {
+        try {
+            XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
+            builder.map(source);
+            indexSettings(Strings.toString(builder), builder.contentType());
+        } catch (IOException e) {
+            throw new ElasticsearchGenerationException("Failed to generate [" + source + "]", e);
+        }
+        return this;
+    }
+
+    /**
+     * Returns settings that should be added/changed for restored index
+     */
+    public Settings indexSettings() {
+        return this.indexSettings;
+    }
+
+    /**
+     * Parses mount definition
+     *
+     * @param source mount definition
+     * @return this request
+     */
+    @SuppressWarnings("unchecked")
+    public MountSearchableSnapshotRequest source(Map<String, Object> source) {
+        // TODO: this could be rewritten with a Constructing Object Parser rather than manually
+        //  parsing the XContent, this was copied/modified from RestoreSnapshotRequest
```

Comment:
I'd prefer we did this TODO in this PR, since I think it would mean we could drop the need for validation and make all the required fields into constructor args (and final) instead.